android/src/main/java/com/tailscale/ipn/ui/model/Ipn.kt

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2024 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package com.tailscale.ipn.ui.model
+
+import kotlinx.serialization.*
+
+class ipn {
+
+    // Represents the overall state of the Tailscale engine.
+    enum class State(val value: Int) {
```

Comment:
I know on Windows there's this extra state on top of this that expresses "UI state," and includes a "needsAdminSigning" state for tailnet lock. I'm curious how this is done on other platforms

Reply:
That's handled differently on iThings.  There's a separate set of localAPI calls for it.   

I'm going to add one API at a time along with the models it needs.  I suspect we'll have to revisit those as we start to actually use them to drive UI.  I'm going to use the iOS version as the basis here, since it's going to functionally be the closest in terms of language, idioms, and general functionality. I want to keep the commits a reasonable size though.